marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/DeployUserAmpsTest.java

```diff
@@ -0,0 +1,137 @@
+package com.marklogic.hub.dhs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.impl.HubConfigImpl;
+import com.marklogic.hub.impl.Versions;
+import com.marklogic.mgmt.ManageClient;
+import com.marklogic.mgmt.ManageConfig;
+import com.marklogic.mgmt.resource.security.AmpManager;
+import com.marklogic.mgmt.resource.security.RoleManager;
+import com.marklogic.mgmt.resource.security.UserManager;
+import com.marklogic.rest.util.ResourcesFragment;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.nio.file.Path;
+
+
+public class DeployUserAmpsTest extends AbstractHubCoreTest {
+    ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void checkIfTestsCanBeRun(){
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assertions.assertTrue(mlVersion.isNightly() || (mlVersion.getMajor() > 10) ||(mlVersion.getMajor() == 10 && mlVersion.getMinor() >= 440));
+    }
+
+    @Test
+    void createAmpWithInheritableRole() {
```

Comment:
For this test, we want to make sure that the amp works too. To do that, create a new "user-amps" test project and put a single XQuery library in there (since those are easier to amp). Something like this:

```
declare function get-doc($uri as xs:string) { fn:doc($uri) }
```

Then insert a document that e.g. has data-hub-developer for read/update permissions. The amp will grant data-hub-developer.

As a data-hub-operator, first make sure that the user gets back null when calling that function on the test document. Then deploy the amp as a dhsa. Then as a dho, call the function again and verify that you get the test document back.

Reply:
Created an amp for js library